utils/test-utils/src/main/java/com/microsoft/device/dualscreen/testing/runner/FoldableJUnit4ClassRunner.kt

```diff
@@ -0,0 +1,54 @@
+/*
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License.
+ */
+
+package com.microsoft.device.dualscreen.testing.runner
+
+import androidx.test.internal.runner.junit4.AndroidJUnit4ClassRunner
+import androidx.test.internal.util.AndroidRunnerParams
+import androidx.test.platform.app.InstrumentationRegistry
+import androidx.test.uiautomator.UiDevice
+import com.microsoft.device.dualscreen.testing.DeviceModel
+import com.microsoft.device.dualscreen.testing.filters.DualScreenTest
+import com.microsoft.device.dualscreen.testing.filters.SingleScreenTest
+import com.microsoft.device.dualscreen.testing.filters.TargetDevice
+import com.microsoft.device.dualscreen.testing.isSurfaceDuo
+import org.junit.runner.Description
+import org.junit.runner.notification.RunNotifier
+import org.junit.runners.model.FrameworkMethod
+
+/**
+ * A specialized [AndroidJUnit4ClassRunner] that can be used together with the
+ * [SingleScreenTest], [DualScreenTest] and [TargetDevice] annotations
+ * to filter the tests that will run on the specified devices.
+ * For example, if a test method is annotated with @TargetDevice(device = DeviceModel.SurfaceDuo),
+ * that test method will run only on SurfaceDuo device or emulator, otherwise will be skipped.
+ */
+class FoldableJUnit4ClassRunner : AndroidJUnit4ClassRunner {
+    constructor(klass: Class<*>?) : super(klass)
+
+    constructor(klass: Class<*>?, runnerParams: AndroidRunnerParams) : super(klass, runnerParams)
+
+    override fun runChild(method: FrameworkMethod?, notifier: RunNotifier?) {
+        val description = describeChild(method)
+        val uiDevice = UiDevice.getInstance(InstrumentationRegistry.getInstrumentation())
+        val targetDevice: TargetDevice? = getAnnotation(description, TargetDevice::class.java)
+        if ((targetDevice == null) ||
+            (!targetDevice.device.isSurfaceDuo && !uiDevice.isSurfaceDuo()) ||
+            (targetDevice.device.isSurfaceDuo && uiDevice.isSurfaceDuo())
```

Comment:
Not sure if I'm interpreting this correctly, but it looks like we're basically using @TargetDevice to limit tests to run on Surface Duo/Surface Duo 2 - to simplify the logic, maybe we can update the annotation to accept a list of multiple target devices so we don't have to make the `isSurfaceDuo()` checks. The updated logic check could maybe look something like this: `if ( targetDevices == null || targetDevices.any { it?.device == uiDevice.getDeviceModel() } )`

If you prefer it as is, I think that also works! Then I think it would be good to add the `DeviceModel.isSurfaceDuo` implementation directly to the DeviceModel class, that way the UiDevice extension could reuse that code instead of having duplicate logic

Reply:
What I wanted here is to skip the test if the test is for surface duo but the device is something else. But yeah, this is a good idea to add there a list of wanted devices. Thanks

Reply:
Related to isSurfaceDuo extension, for now I want to leave it here .In the future I want to distinguish tests that will run on Surface Duo One or Two and most probably this extension will be deleted.

Reply:
Sounds good, thank you!